### PR TITLE
Start health check for every neighbour

### LINKED ISSUE #807: Route discovery will not use the best possible path
## Problem Definition

Route discovery algorithm picks first of the neighbour nodes, sorted by these criteria:

1. node status
2. number of hops to the target
3. funds in the channel

Node status is determined by the healthcheck() greenlet, which periodically tries to send Ping message to the other node. If node replies with an Ack, channel status is set to 'reachable', otherwise 'unknown' is set. Node status is set 'unknown' by default, even without the check running.

However, the greenlet is not started by default for each neighbor, but only for targets of a transfer. Because of this, transfers are routed in the network more or less randomly, depending on targets of previous transfers.

## Solution

Either change the routing algorithm sorting criteria to not prefer node by its status, or start healthcheck greenlet for each neighbor of the node.

## Tasklist

- [x] Discuss which solution to choose
- [x] Change the route planner/healthcheck startup

### PR DESCRIPTION
**Pull request by pcppcp**
To make sure we have complete information to find the shortest route, we start healthcheck for every participant we've created a channel with.

This replaces PR #815 to fix #807.

### CODE REVIEW

File: raiden/network/channelgraph.py [hunk 1]
```diff
@@ -177,6 +177,7 @@ def get_best_routes(
         network_state = nodeaddresses_statuses[partner_address]
         route_state = channel_to_routestate(channel, partner_address)
 
+        log.debug("possible route %s %s %d" % (pex(partner_address), network_state, _))
```
Comment (LefterisJP) [inline raiden/network/channelgraph.py]: Why is the last argument `_` here?

**Comment by LefterisJP** [comment]
I don't see any problem with this but would like @hackaugusto to have a look before merging too.

File: raiden/tests/unit/test_service.py [hunk 2]
```diff
@@ -68,8 +68,14 @@ def test_ping_unreachable(raiden_network):
...
-    for message in messages:
-        assert decode(message) == ping_message
+    # Raiden node will start pinging as soon as a new channel
+    #  is established. We need to test if
+    #  a) there is our original message in the queue
+    #  b) there are only Ping message types in
+    messages_decoded = [decode(m) for m in messages]
+    assert ping_message in messages_decoded
+    for message in messages_decoded:
+        assert isinstance(message, Ping)
```
Comment (hackaugusto) [inline raiden/tests/unit/test_service.py]: This check is unrelated to the unit test "test_ping_unreachable". Please write a new unit test, give it a proper name and if you want a docstring specifying the regression it's fixing.

File: raiden/event_handler.py [hunk 3]
```diff
@@ -245,6 +245,15 @@ def handle_channelnew(self, state_change):
...
             )
+            # This is little hacky, used when bootstraping a token network.
+            # See ConnectionManager for details.
+            have_bootstrap = ((participant1 == connection_manager.BOOTSTRAP_ADDR) or
+                              (participant2 == connection_manager.BOOTSTRAP_ADDR))
+            if not have_bootstrap:
+                self.raiden.start_health_check_for(
+                    participant2 if
+                    participant1 == self.raiden.address else
+                    participant1)
```
Comment (hackaugusto) [inline raiden/event_handler.py]: Humm, two things:
1. This check is really noisy.
2. I don't think we should register a netting channel with the bootstrap address, this was an oversight.

Suggestion, simplify the check and move it above line `243`, e.g.:
```
is_participant = self.raiden.address in (participant1, participant2)
is_bootstrap = connection_manager.BOOTSTRAP_ADDR in (participant1, participant2)
if is_participant and not is_bootstrap:
```

Comment (pcppcp) [inline raiden/event_handler.py]: Are you sure about not registering the bootstrap channel? I thought it was neccessary to have at least one channel registered when creating a new token network, so other participants will discover creator's address.

File: raiden/network/channelgraph.py (hunk 1 above)
Comment (pcppcp) [inline raiden/network/channelgraph.py]: Sorry, this was just a debug print meant to be deleted later.

File: raiden/tests/unit/test_service.py (hunk 2 above)
Comment (pcppcp) [inline raiden/tests/unit/test_service.py]: This change is only fixing this test's behaviour to make it pass. Upon creating the channel, node will start pinging the neighbour immediately and there will be many Ping messages in the queue pending.
The original test expects only a single ping message, created from within the test, so I had to extend the checks.

File: raiden/event_handler.py (hunk 3 above)
Comment (hackaugusto) [inline raiden/event_handler.py]: > I thought it was neccessary to have at least one channel registered when creating a new token network,

Yes, this is required. The connection manager is the piece of code that will open connections to new channels, it will kick off on user demand and create a channel *on-chain* for starting the network.

The code that this PR is patching is for the python counter part, that is not really needed since BOOTSTRAP_ADDR is assumed to not exist.

Comment (hackaugusto) [inline raiden/event_handler.py]: Although, having the python channel is not bad either, so do whatever is the simplest, but please, make the conditionals clearer.
